Add single reconstruction option

thomasmfish opened the pull request:
This allows more specific processing path definition for single reconstructions, along with custom multiprocessing pool definition.

Depends on #88

PeterC-DLS commented on src/sim_recon/recon.py:
```diff
@@ -364,37 +364,39 @@ def _get_incomplete_channels(
-        multiprocessing.Pool(
+    if multiprocessing_pool is None:
+        pool = multiprocessing.Pool(
             processes=1 + int(parallel_process),  # 2 processes max
```
```suggestion
            processes=2 if parallel_process else 1,  # 2 processes max
```

PeterC-DLS commented on src/sim_recon/recon.py:
```diff
@@ -364,37 +364,39 @@ def _get_incomplete_channels(
+        logging_redirect = get_logging_redirect()
+        progress_wrapper = get_progress_wrapper()
+
+        # `maxtasksperchild=1` is necessary to ensure the child process is cleaned
```
Comment now pertains to location several lines above

PeterC-DLS reviewed (approved):
LGTM

Commit:
Split out `run_single_reconstruction` from `run_reconstructions`

Update pool creation

Commit:
Create `sim_reconstruct_single`

Has ability to set processing directory directly (not the parent directory)
Can set the multiprocessing pool to be used
